test/README.md

```diff
@@ -0,0 +1,205 @@
+
+For example, `tests/utc/run.sh` consists of:
+
+```bash
+#!/bin/bash
+set -e
+
+docker run --rm --entrypoint date "$1" +%Z
+```
+
+Which runs `date` inside the given image, and outputs the current timezone according to `date +%Z`, which should be `UTC`. See `tests/<test name>/expected-std-out.txt` below for the other half of this test.
+
+There are several pre-written `run.sh` scripts to help with common tasks. See the `tests/run-*-in-container.sh` section below for more details about these.
+
+#### `tests/<test name>/expected-std-out.txt`
+
+If this file exists, then this test's `run.sh` is expected to output exactly the contents of this file on standard output, and differences will be assumed to denote test failure.
+
+To continue with our `utc` test example, its contents in this file are simply:
+
+	UTC
+
+If a test fails due to having incorrect output, the `diff` between the generated output and the expected output will be displayed:
+
+```console
+$ ./run.sh -t utc fedora:21
+testing fedora:21
+	'utc' [1/1]...failed; unexpected output:
+--- tests/utc/expected-std-out.txt	2015-02-05 16:52:05.013273118 -0700
++++ -	2015-03-13 15:11:49.725116569 -0600
+@@ -1 +1 @@
+-UTC
++EDT
+```
+
+(this is an example of `fedora:21` failing the `utc` test)
+
+#### `tests/run-*-in-container.sh`
+
+These scripts are intended to be used as symlink targets for test `run.sh` scripts.
+
+For example, if `tests/python-sometest/run.sh` is a symlink to `../run-python-in-container.sh`, then `tests/run-python-in-container.sh` will execute `python tests/python-sometest/container.py` inside the image.
+
+To illustrate further:
+
+```console
+$ ls -l tests/hylang-sh/
+total 8
+-rw-rw-r-- 1 tianon tianon 167 Mar 11 13:24 container.hy
+-rw-rw-r-- 1 tianon tianon   5 Mar 11 13:24 expected-std-out.txt
+lrwxrwxrwx 1 tianon tianon  25 Mar 11 13:25 run.sh -> ../run-hy-in-container.sh
+$ ls -l tests/python-hy/
+total 4
+lrwxrwxrwx 1 tianon tianon 25 Mar 11 13:25 container.hy -> ../hylang-sh/container.hy
+-rw-rw-r-- 1 tianon tianon 56 Mar 11 13:24 container.sh
+lrwxrwxrwx 1 tianon tianon 33 Mar 11 13:25 expected-std-out.txt -> ../hylang-sh/expected-std-out.txt
+lrwxrwxrwx 1 tianon tianon 27 Feb  5 16:52 run.sh -> ../run-bash-in-container.sh
+$ cat tests/python-hy/container.sh
+#!/bin/bash
+set -e
+
+pip install -q hy
+hy ./container.hy
+```
+
+## Configuring the Test Suite
+
+### `config.sh`
+
+This file controls which tests apply (or don't apply) to each image.
+
+When testing an image, the list of tests to apply are calculated by doing `globalTests + imageTests[testAlias[image]] + imageTests[image]`. Any tests for which `globalExcludeTests[image_test]` or `globalExcludeTests[image:variant_test]` is set are removed. If `-t` is specified one or more times, any tests not specified explicitly via `-t` are also removed.
```

Comment:
Do we need to mention `+ imageTests[testAlias[image]:variant] + imageTests[image:variant]` here?